### PR TITLE
travis: Get CMake from kitware repo

### PR DESCRIPTION
**Pull request by Patater**
### Description

Pypi doesn't always have the lastest CMake releases, so use the official
Kitware apt repo instead.

### Test Coverage

- [X]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -41,6 +41,7 @@ addons:
     sources:
       - sourceline: 'deb https://apt.kitware.com/ubuntu/ xenial main'
         key_url: 'https://apt.kitware.com/keys/kitware-archive-latest.asc'
+      - sourceline: 'deb https://apt.kitware.com/ubuntu/ xenial-rc main'
```
Comment (jainvikas8) [inline .travis.yml:44]: Are we planning to use pre-release versions in the near future? If this is a temporary change then might be good to raise an issue on Github and remove this when we don't need it

Comment (Patater) [reply .travis.yml:44]: I am not sure what Mbed OS future plans are. When Mbed OS limits themselves to only released versions, we can drop this, yes.

Comment (Patater) [reply .travis.yml:44]: https://github.com/ARMmbed/mbed-tools/issues/104

**Review by jainvikas8 — APPROVED** [review]

**Review by jainvikas8 — APPROVED** [review]
Please change commit message
`Make is easier to inspect what command was run`

Ok to merge, if Travis is happy

**Review by jainvikas8 — APPROVED** [review]

**Pull request merged**